.buildkite/publish/publish-common.sh

```diff
@@ -18,11 +18,34 @@ source $SCRIPT_DIR/git-setup.sh
 VERSION_PATH="$PROJECT_ROOT/connectors/VERSION"
 export VERSION=$(cat $VERSION_PATH)
 
+# Variables for build.yaml
+version="${VERSION}"
+version_qualifier="${VERSION_QUALIFIER:-}"
+revision="$(git rev-parse HEAD)"
+repository="$(git config --get remote.origin.url)"
+
 if [[ "${USE_SNAPSHOT:-}" == "true" ]]; then
+  # SNAPSHOT workflows ignore version qualifier
   echo "Adding SNAPSHOT labeling"
   export VERSION="${VERSION}-SNAPSHOT"
+  version="${VERSION}"
+elif [[ -n "${VERSION_QUALIFIER:-}" ]]; then
+  # version qualifier should not append version in build.yaml
+  echo "Adding version qualifier labeling"
+  version="${VERSION}-${VERSION_QUALIFIER}"
 fi
 
+# Create a build.yaml file for reference after build process
+cat <<EOL > connectors/build.yaml
```

Comment:
This file will end up being written every time a script calls `source publish-common.sh`. I think that's ok, but wanted to call it out in case that wasn't intended, and it might have negative side effects. Fore example, this file is sourced during the DRA publish steps, not just during the image building steps

Reply:
I hadn't considered this, so that's actually unintentional. If we decided that we do care about that, do you think this should be moved to another file, so we can control its importing better?

Reply:
I think this logic _could_ be moved to `build-docker.sh`, but I don't see any obvious risks with where it is. If you don't see it hurting anything either, I'm find to leave it here for now.